core/jdiameter/api/src/main/java/org/jdiameter/api/sy/ClientSySession.java

```diff
@@ -0,0 +1,78 @@
+  * GNU Affero General Public License for more details.
+  *
+  * You should have received a copy of the GNU Affero General Public License
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>
+  *
+  * This file incorporates work covered by the following copyright and
+  * permission notice:
+  *
+  *   JBoss, Home of Professional Open Source
+  *   Copyright 2007-2011, Red Hat, Inc. and individual contributors
+  *   by the @authors tag. See the copyright.txt in the distribution for a
+  *   full listing of individual contributors.
+  *
+  *   This is free software; you can redistribute it and/or modify it
+  *   under the terms of the GNU Lesser General Public License as
+  *   published by the Free Software Foundation; either version 2.1 of
+  *   the License, or (at your option) any later version.
+  *
+  *   This software is distributed in the hope that it will be useful,
+  *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+  *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  *   Lesser General Public License for more details.
+  *
+  *   You should have received a copy of the GNU Lesser General Public
+  *   License along with this software; if not, write to the Free
+  *   Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  *   02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  */
+
+package org.jdiameter.api.sy;
+
+import org.jdiameter.api.IllegalDiameterStateException;
+import org.jdiameter.api.InternalException;
+import org.jdiameter.api.OverloadException;
+import org.jdiameter.api.RouteException;
+import org.jdiameter.api.app.AppSession;
+import org.jdiameter.api.app.StateMachine;
+
+import org.jdiameter.api.sy.events.SpendingLimitRequest;
+import org.jdiameter.api.sy.events.SpendingTerminationRequest;
+import org.jdiameter.api.sy.events.SpendingNotificationAnswer;
+
+/**
+ * Basic class for Sy Client Interface specific session.
+ * Listener must be injected from constructor of implementation class.
+ *
+ * @author <a href="mailto:aferreiraguido@gmail.com"> Alejandro Ferreira Guido </a>
+ * @author <a href="mailto:brainslog@gmail.com"> Alexandre Mendonca </a>
+ * @author <a href="mailto:baranowb@gmail.com"> Bartosz Baranowski </a>
+ */
+public interface ClientSySession extends AppSession, StateMachine {
+
+  void sendInitialSpendingLimitReportRequest(SpendingLimitRequest request)
+      throws InternalException, IllegalDiameterStateException, RouteException, OverloadException;
+
+  void sendIntermediateSpendingLimitReportRequest(SpendingLimitRequest request)
+      throws InternalException, IllegalDiameterStateException, RouteException, OverloadException;
+
+  void sendFinalSpendingLimitReportRequest(SpendingTerminationRequest request)
+      throws InternalException, IllegalDiameterStateException, RouteException, OverloadException;
+
+  void sendSpendingLimitReportAnswer(SpendingNotificationAnswer answer)
```

Comment:
Shouldn't this method be `sendSpendingStatusNotificationAnswer` ?
Also, the event class name should also be `SpendingStatusNotificationAnswer` instead of `SpendingNotificationAnswer`, IMO, to be according with the spec message name.